registry/storage/filewriter_test.go

```diff
@@ -165,8 +165,9 @@ func TestSimpleWrite(t *testing.T) {
 
 func TestBufferedFileWriter(t *testing.T) {
 	ctx := context.Background()
-	writer, err := newFileWriter(ctx, inmemory.New(), "/random")
 
+	writerp, err := newFileWriter(ctx, inmemory.New(), "/random")
+	writer := *writerp
```

Comment:
@stevvooe 
A bufferedFileWriter is dereference in the code at: https://github.com/aibaars/distribution/blob/9f83bfc071094019b942c4332d17327b4dc5cdfc/registry/storage/linkedblobstore.go#L200

This dereference bufferedFileWriter behaves in the same way as the one in the test case. To trigger a failure you need to remove the ```ReadStream``` interface (to make sure the code actually calls the ```Write``` methods) in :
```
+++ b/blobs.go
@@ -154,7 +154,6 @@ type BlobIngester interface {
 // recovered with the id.
 type BlobWriter interface {
        io.WriteSeeker
+ //   io.ReaderFrom
        io.Closer
```
and run ```go test  ./registry/handlers/...```

this fails the handlers test suite with:

```
--- FAIL: TestBlobAPI (1.71s)
	api_test.go:1472: map... Range header value not matched in response: "0-0" != "0-17191423"
```
As you can see the offset in the range is zero instead of the expected large number.

I think you can agree that the test cases should not fail when you switch between ```ReadFrom``` or buffered ```Writer```.